circuit_maintenance_parser/data.py

```diff
@@ -77,7 +77,8 @@ def init_from_emailmessage(cls: Type["NotificationData"], email_message) -> Opti
             # Adding extra headers that are interesting to be parsed
             data_parts.add(DataPart(EMAIL_HEADER_SUBJECT, email_message["Subject"].encode()))
             data_parts.add(DataPart(EMAIL_HEADER_DATE, email_message["Date"].encode()))
-            return cls(data_parts=list(data_parts))
+            # Ensure the data parts are processed in a consistent order
+            return cls(data_parts=sorted(data_parts, key=lambda part: part.type))
```

Comment:
why order is important?

Reply:
In the case where we have information of different accuracy/precision available from different data parts. For example with Zayo we can always pull an (approximate) `stamp` from the email date, but for some notifications we may have a more precise `stamp` explicitly stated in the email body. Without this change it was unpredictable as to which one of these would be processed last and therefore win out - see the CI failures on 7970ecc above for an example of this.

Reply:
I understand, good catch.